### PR TITLE
fix(vua-app): clone mount error to prevent mutating read-only error object

### LINKED ISSUE #5056: Attempted to assign to readonly property.
### Version

[v2.4.3](https://github.com/nuxt.js/releases/tag/v2.4.3)

### Reproduction link

[https://github.com/nuxt/nuxt.js/blob/f791d786e0996e4cad2b1ddbe244a747e7e700aa/packages/vue-app/template/client.js#L81](https://github.com/nuxt/nuxt.js/blob/f791d786e0996e4cad2b1ddbe244a747e7e700aa/packages/vue-app/template/client.js#L81)

### Steps to reproduce

Throw a error while mounting Nuxt (client)

### What is expected ?

To adjust the error message

### What is actually happening?

A new error is thrown saying: `Attempted to assign to readonly property.`

### Additional comments?

https://github.com/nuxt/nuxt.js/blob/f791d786e0996e4cad2b1ddbe244a747e7e700aa/packages/vue-app/template/client.js#L81

<div align="right"><sub><em>This bug report is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/nuxt.js/issues/c8681">#c8681</a>)</em></sub></div>

### PR DESCRIPTION
**Pull request by TheAlexLichter**
## Types of changes
- [x] Bug fix (a non-breaking change which fixes an issue)

Resolves #5056 

## Checklist:
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [ ] I have added tests to cover my changes (if not applicable, please state why)
- [x] All new and existing tests are passing.

### CODE REVIEW

File: packages/vue-app/template/client.js [hunk 1]
```diff
@@ -81,8 +81,10 @@ const errorHandler = Vue.config.errorHandler || console.error
 createApp()
   .then(mountApp)
   .catch((err) => {
-    err.message = '[nuxt] Error while mounting app: ' + err.message
-    errorHandler(err)
+    const improvedErr = new Error('[nuxt] Error while mounting app: ' + err.message)
+    improvedErr.original = err
+    improvedErr.stack = improvedErr.stack.split('\n').slice(0,2).join('\n') + '\n' + err.stack
```
Comment (aldarund) [inline packages/vue-app/template/client.js]: I think stack might be undefined.
```
x = new DOMException()
DOMException
x.stack
undefined
```

Comment (aldarund) [inline packages/vue-app/template/client.js]: When err.stack undefined will it look good?

Comment (pi0) [inline packages/vue-app/template/client.js]: This is a very low-level hack which may make error handler even more unstable for different browsers / errors

Comment (manniL) [reply packages/vue-app/template/client.js]: Good catch!

Comment (manniL) [reply packages/vue-app/template/client.js]: Could be, yeah. But What would be another solution? Checking for a DOMException (and other errors with read-only values)? :thinking:

**Comment by pi0** [comment]
@manniL Isn't `new Error(error)` enough to workaround readonly object?

File: packages/vue-app/template/client.js (hunk 1 above)
Comment (aldarund) [reply packages/vue-app/template/client.js]: Another thing is that in IE for example improvedErr.stack would be undefined and crash
```
new Error().stack
undefined
```

**Comment by manniL** [comment]
Welp, that was too simple :joy:

You are right @pi0 :relaxed:

**Review by aldarund — APPROVED** [review]

**Review by pi0 — APPROVED** [review]

**Pull request merged**